metrics/config.go

```diff
@@ -67,10 +67,9 @@ func getMetricsConfig(m map[string]string, domain string, component string, logg
 
 	if mc.backendDestination == Stackdriver {
 		sdProj, ok := m[stackdriverProjectIDKey]
-		if !ok || sdProj == "" {
-			return nil, errors.New("For backend stackdriver, metrics.stackdriver-project-id field must exist and cannot be empty")
+		if ok && sdProj != "" {
+			mc.stackdriverProjectID = sdProj
```

Comment:
When this is left empty, do OpenCensus pick the current GKE project as the SD project id?

Reply:
The question is my mind is: how discoverable would that be for users?

Noting that `metadata.ProjectID()` returns an optional `error` (e.g. not running on GCE), I would contend that we should explicitly use it here and surface the error (rejecting the configmap update).

Reply:
@mdemirhan yes, OpenCensus is able to use the current GCP project as the SD project ID in my testing.

@mattmoor  If OpenCensus is not able to use the default account, it'll return an error when creating a metrics exporter: 
errors.New("stackdriver: no project found with application default credentials")"
Is this sufficient?

Reply:
This error would surface [here](https://github.com/knative/pkg/pull/147/files#diff-0f986afc14030fe191c861028eeb004dR105)?

Reply:
Yes.

Reply:
Maybe add a comment for what will happen in case stackdriverProjectIDKey is not provided and OpenCensus is not able to use the default account.

Reply:
Added a comment